src/main/java/io/opentracing/thrift/SpanProtocol.java

```diff
@@ -123,4 +128,22 @@ public void readMessageEnd() throws TException {
       }
     }
   }
+
+  /**
+   * A {@link TProtocolFactory} that makes sure the returned {@link TProtocol} instances are
+   * wrapped with a {@link SpanProtocol}.
+   */
+  public static class Factory implements TProtocolFactory {
```

Comment:
Why it is static class? Why not put it in own file?

Reply:
Mostly to match what the Thrift library does with `TBinaryProtocol.Factory`.